test/functional/toolchains/meson/test_cross_compilation.py

```diff
@@ -244,3 +244,34 @@ def test_use_meson_toolchain():
     # Build locally
     c.run('build . --profile:host=android')
     assert "conanfile.py (hello/0.1): Calling build()" in c.out
+
+
+@pytest.mark.tool("meson")
+def test_baremetal_cross_compiling():
```

Comment:
Most likely this test should be restricted to Linux only, the configuration for ``gcc`` in other plaforms require more configuration.